### PR TITLE
End game gui implementation and subsequent fixes

### PR DESCRIPTION
**Pull request by hippi345**
still needs the gui to be intuitive and the end game conditions to scale to multiple game scenarios

### CODE REVIEW

File: src/sample/Constants.java [hunk 1]
```diff
@@ -7,4 +7,6 @@
 
     // colors
     static String LIGHT_BLUE = "-fx-background-color:#00BFFF; -fx-opacity:1;";
+
+    static  boolean GAME_COMPLETE = false;
```
Comment (RiskRunner0) [inline src/sample/Constants.java:9]: Looks like 2 spaces

File: src/sample/Game.java [hunk 2]
```diff
@@ -29,7 +30,7 @@ void InitializeGame(View gameView)
         {
             this.bot.RandomizeLocation(gameView.gridSize);
         }
-        gameView.adjustBotAndTreasureLocations(this.bot, this.treasure);
+        // gameView.redrawGrid(this.bot, this.treasure);
```
Comment (RiskRunner0) [inline src/sample/Game.java]: If we don't need this, you can remove it.

File: src/sample/Game.java [hunk 3]
```diff
@@ -77,51 +59,19 @@ private boolean treasureIsFound()
...
-        Main.gameView.adjustBotAndTreasureLocations(this.bot, this.treasure);
+        Main.gameView.redrawGrid(this.bot, this.treasure);
     }
 
     // treasure detection
     private void treasureDetection()
     {
-        if(treasureIsFound())
+        if(treasureIsFound()) {
+            Constants.GAME_COMPLETE = true;
```
Comment (RiskRunner0) [inline src/sample/Game.java]: Setting a constant is counter intuitive, if you're modifying it, it's not constant.  Maybe make this a field on the game?

**Review by RiskRunner0 — CHANGES_REQUESTED** [review]

File: src/sample/Game.java (hunk 3 above)
Comment (hippi345) [reply src/sample/Game.java]: Autoplay being implemented in Main would make this bit involved but I could make it public and proceed on that?

File: src/sample/Game.java (hunk 2 above)
Comment (hippi345) [reply src/sample/Game.java]: fairsies enough

File: src/sample/Constants.java (hunk 1 above)
Comment (hippi345) [reply src/sample/Constants.java:9]: wdym?

File: src/sample/Game.java (hunk 3 above)
Comment (RiskRunner0) [reply src/sample/Game.java]: Main is where the game is created, so you could do like ```Game.GameComplete``` in there, and then set that variable on the game here.

File: src/sample/Constants.java (hunk 1 above)
Comment (RiskRunner0) [reply src/sample/Constants.java:9]: Between static and boolean, I think there's 2 spaces

Comment (hippi345) [reply src/sample/Constants.java:9]: was separating those intentionally since they are different things but I am resolving

**Comment by hippi345** [comment]
@QueenCitySmitty  could you give this one a run through with comments
I think some of the things considered needing fixed may have been addressed in one of your more recent PR's - so feel free to let me know of things in this PR that I can fix (short of needing a separate PR) as I don't know what has been addressed since this PR 2 days ago but I want to make sure it is sound to merge

**Comment by RiskRunner0** [comment]
I'm not seeing anything about the end game gui in this pr?

**Comment by hippi345** [comment]
@QueenCitySmitty 
Give this one a run and see if the execution is ok for a merge - I put the end game gui in the View class for sake of organization and implemented the endgame actions within your latest restructure for the Game so Game calls -> View to close the game and open the end game gui

File: src/sample/newThreadItems.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package sample;
+
+public class newThreadItems implements Runnable
```
Comment (RiskRunner0) [inline src/sample/newThreadItems.java]: Do we need this class right now for this PR?  I don't see it being used anywhere

File: src/sample/Game.java [hunk 5]
```diff
@@ -86,6 +86,11 @@ private void executeTurn(MoveEnum botMovementDirection)
 
         DetermineCurrentStatus();
         Main.gameView.adjustBotAndTreasureLocations(previousBotPoint, this.bot, this.treasure);
+        if (_status == GameStatusEnum.Complete)
+        {
+            System.out.println("You found the treasure!");
```
Comment (RiskRunner0) [inline src/sample/Game.java:91]: We should probably expose status, and read from it after a move instead, we might need to use it later.

**Review by RiskRunner0 — CHANGES_REQUESTED** [review]

File: src/sample/newThreadItems.java (hunk 4 above)
Comment (hippi345) [reply src/sample/newThreadItems.java]: Resolved-this is in another commit external to this one.

File: src/sample/Game.java (hunk 5 above)
Comment (hippi345) [reply src/sample/Game.java:91]: wdym? 
for context, I did not make status and this change in the determination to end the game.
So I am not sure what you are asking on this comment. 
What do you mean by "expose"? Save it in memory? Output it to the console? Etc.

Comment (hippi345) [reply src/sample/Game.java:91]: @QueenCitySmitty

Comment (RiskRunner0) [reply src/sample/Game.java:91]: Expose it as make it a public field that anything can access, like at any time I should be able to query the game for the current status.

**Pull request merged**